src/mat73.c

```diff
@@ -651,6 +651,9 @@ Mat_H5ReadFieldNames(matvar_t *matvar, hid_t dset_id, hsize_t *nfields)
     } else {
         err = MATIO_E_NO_ERROR;
     }
+    if ( *nfields == 0 ) {
+        *nfields = 1;
```

Comment:
I actually wonder if we should return with error (MATIO_E_OPERATION_NOT_SUPPORTED or MATIO_E_GENERIC_READ_ERROR) instead.